File: src/main/java/com/travel/global/result/ResultType.java
//package com.travel.global.result;
//
//import com.travel.global.constants.CommonConstants;
//
//import java.util.HashMap;
//import java.util.List;
//import java.util.Map;
//
//public class ResultType {
//    public static Map<String, Object> resultList(List<?> list, Long listCount) {
//        Map<String, Object> resultMap = new HashMap<>();
//        resultMap.put(CommonConstants.RESULT_LIST, list);
//        resultMap.put(CommonConstants.RESULT_DATA, listCount);
//
//        return resultMap;
//    }
//
//    public static Map<String, Object> resultData(Object object) {
//        Map<String, Object> resultMap = new HashMap<>();
//        resultMap.put(CommonConstants.RESULT_DATA, object);
//
//        return resultMap;
//    }
//}

File: src/main/kotlin/com/travel/global/result/ResultType.kt
package com.travel.global.result

import com.travel.global.constants.CommonConstants

object ResultType {
    fun resultList(list: List<*>, listCount: Long?): Map<String, Any> {
        val resultMap: MutableMap<String, Any> = HashMap()
        resultMap[CommonConstants.RESULT_LIST] = list
        resultMap[CommonConstants.RESULT_DATA] = listCount ?: 0L
        return resultMap
    }

    fun resultData(obj: Any): Map<String, Any> {
        val resultMap: MutableMap<String, Any> = HashMap()
        resultMap[CommonConstants.RESULT_DATA] = obj
        return resultMap
    }
}
